### PR TITLE
feat: Do not apply shared colors to non-group series

### PR DESCRIPTION
**Pull request by filp**
Closes: #8978 

### Description:

Skips the shared color assignment logic for non-group series, instead using the default behavior that applies color from the palette in order.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/UserSettings/AppearanceSettingsPanel/index.tsx [hunk 1]
```diff
@@ -47,15 +48,8 @@ const defaultColorPalettes: ColorPalette[] = [
...
-            '#ff8933',
-            '#ffbd33',
-            '#ffe133',
-            '#e8ff33',
-            '#b4ff33',
-            '#80ff33',
-            '#4cff33',
-            '#33ff49',
+            // Use the initial 9 colors directly from ECHARTS to keep them in sync:
+            ...ECHARTS_DEFAULT_COLORS,
```
Comment (filp) [inline packages/frontend/src/components/UserSettings/AppearanceSettingsPanel/index.tsx:52]: Not really related to this task, but makes the provenance of those initial 9 colors clearer, and avoids things going out of sync in the future.

File: packages/frontend/src/hooks/useChartColorConfig.ts [hunk 2]
```diff
@@ -5,6 +5,13 @@ import { EChartSeries } from './echarts/useEchartsCartesianConfig';
 
 export type SeriesLike = EChartSeries | Series;
 
+export const isGroupedSeries = (series: SeriesLike) => {
```
Comment (filp) [inline packages/frontend/src/hooks/useChartColorConfig.ts:8]: Has a bit of duplication with the logic below, but opted to keep it like this for clarity.

**Comment by filp** [comment]
@magnew Basically color sharing will happen only in cases where there are distinct groups (so pie charts, or other charts with a grouping dimension). This means most charts will follow predictably the order of colors as they are defined in the palette.

If there's a grouping dimension, shared color assignment kicks in and attempts to give each group a deterministic color value.

**Comment by magnew** [comment]
> @magnew Basically color sharing will happen only in cases where there are distinct groups (so pie charts, or other charts with a grouping dimension). This means most charts will follow predictably the order of colors as they are defined in the palette.
> 
> If there's a grouping dimension, shared color assignment kicks in and attempts to give each group a deterministic color value.

Cool, thanks for explaining. I liked when the colors got shared across dashboards, but I guess people dont like change 😄

**Pull request merged**